src/io/result_io.jl

```diff
@@ -400,54 +400,51 @@ function setup_result_store!(domain::Domain, scen_spec::DataFrame)::Tuple
         return (dl...,)
     end
 
-    met_names = [
-        :relative_cover,
-        :relative_shelter_volume,
-        :absolute_shelter_volume,
-        :relative_juveniles,
-        :juvenile_indicator,
-        :coral_evenness
+    outcome_metrics::Vector{metrics.Metric} = [
+        metrics.relative_cover,
+        metrics.relative_shelter_volume,
+        metrics.absolute_shelter_volume,
+        metrics.relative_juveniles,
+        metrics.juvenile_indicator,
+        metrics.coral_evenness,
+        metrics.relative_taxa_cover
     ]
 
-    _unique_loc_ids = unique_loc_ids(domain)
-
-    dim_struct = Dict(
-        :structure => string.((:timesteps, :locations, :scenarios)),
-        :unique_loc_ids => _unique_loc_ids
+    metric_symbols::Vector{Symbol} = metrics.to_symbol.(outcome_metrics)
+    metric_names::Vector{String} = metrics.to_string.(outcome_metrics; is_titlecase=true)
+    metric_units::Vector{String} = [metric.unit for metric in outcome_metrics]
+    axes_names::Vector{NTuple{3,Symbol}} = fill(
+        (:timesteps, :locations, :scenarios), length(outcome_metrics) - 1
     )
-    result_dims::Tuple{Int64,Int64,Int64} = dim_lengths(dim_struct[:structure])
+    push!(axes_names, (:timesteps, :species, :scenarios))
```

Comment:
Add a comment explaining why this last one is slightly different (I can infer it, but I know the context here, others likely won't)

Reply:
Done